src/lib.rs

```diff
@@ -15,6 +15,10 @@ use std::{
     },
 };
 
+#[cfg(all(unix, feature = "auto-splitting"))]
+#[cfg(unix)]
+use std::thread;
```

Comment:
I'd remove this import, you can just do `std::thread::spawn` down there. No need to write 3 lines to get rid of just `std::`.